src/hotspot/share/gc/shared/ptrQueue.cpp

```diff
@@ -56,11 +56,38 @@ void BufferNode::deallocate(BufferNode* node) {
   FREE_C_HEAP_ARRAY(char, node);
 }
 
+BufferNode::Allocator::PendingList::PendingList() :
+  _tail(nullptr), _head(nullptr), _count(0) {}
+
+BufferNode::Allocator::PendingList::~PendingList() {
+  delete_list(Atomic::load(&_head));
+}
+
+size_t BufferNode::Allocator::PendingList::add(BufferNode* node) {
+  assert(node->next() == nullptr, "precondition");
+  BufferNode* old_head = Atomic::xchg(&_head, node);
+  if (old_head != nullptr) {
+    node->set_next(old_head);
+  } else {
+    assert(_tail == nullptr, "invariant");
+    _tail = node;
+  }
+  return Atomic::add(&_count, size_t(1));
+}
+
+BufferNodeList BufferNode::Allocator::PendingList::take_all() {
+  BufferNodeList result{Atomic::load(&_head), _tail, Atomic::load(&_count)};
+  Atomic::store(&_head, (BufferNode*)nullptr);
+  _tail = nullptr;
+  Atomic::store(&_count, size_t(0));
+  return result;
+}
+
 BufferNode::Allocator::Allocator(const char* name, size_t buffer_size) :
   _buffer_size(buffer_size),
-  _pending_list(),
+  _pending_lists{},
```

Comment:
```suggestion
  _pending_lists(),
```

Some buggy compilers attempt to initialize the aggregate using copy initialization ({ initializer-list }), which consequently calls the deleted PendingList copy constructor. This is a compiler bug; according to the c++14 standard (8.5.1/7 and 8.5.4/3.4), the object should be value-initialized initialized in this case as the initializer list has no elements.

Reply:
Thanks for discovering the problem and the discussion that led to this solution.  I've added that fix to the PR.  In case any readers are interested, the failure was encountered with gcc 8.2.  No idea of the actual version range with the problem.  I spent some time perusing the gcc bug database but didn't find a match there, though that could easily be a failure to find the right search terms.